src/Http/Request.php

```diff
@@ -0,0 +1,332 @@
+      return $_POST;
+    }
+
+    if ($query_string = file_get_contents('php://input')) {
+      // When trying to POST using curl on simpleTest it doesn't reach
+      // $_POST, so we try to re-grab it here.
+      // Also, sometimes the client might send the input still encoded.
+      if ($decoded_json = drupal_json_decode($query_string)) {
+        return $decoded_json;
+      }
+      else {
+        parse_str($query_string, $body);
+        return $body;
+      }
+    }
+
+    return NULL;
+  }
+
+  /**
+   * Parses the header names and values from globals.
+   *
+   * @return HttpHeaderBag
+   *   The headers.
+   */
+  protected static function parseHeadersFromGlobals() {
+    $bag = new HttpHeaderBag();
+    $headers = array();
+    if (function_exists('apache_request_headers')) {
+      $headers = apache_request_headers();
+    }
+    else {
+      foreach ($_SERVER as $key => $value) {
+        if (strpos($key, 'HTTP_') === 0) {
+          // Generate the plausible header name based on the $name.
+          // Converts 'HTTP_X_FORWARDED_FOR' to 'X-Forwarded-For'
+          $name = substr($key, 5);
+          $parts = explode('_', $name);
+          $parts = array_map('strtolower', $parts);
+          $parts = array_map('ucfirst', $parts);
+          $name = implode('-', $parts);
+          $headers[$name] = $value;
+        }
+      }
+    }
+    // Iterate over the headers and bag them.
+    foreach ($headers as $name => $value) {
+      $bag->bag(HttpHeader::create($name, $value));
+    }
+    return $bag;
+  }
+
+  /**
+   * Gets the fully qualified URL with the query params.
+   *
+   * @return string
+   *   The URL.
+   */
+  public function href() {
+    return url($this->path, array(
+      'absolute' => TRUE,
+      'query' => $this->query,
+    ));
+  }
+
+  /**
+   * Get the credentials based on the $_SERVER variables.
+   *
+   * @return array
+   *   A numeric array with the username and password.
+   */
+  protected static function getCredentials() {
+    $username = empty($_SERVER['PHP_AUTH_USER']) ? NULL : $_SERVER['PHP_AUTH_USER'];
+    $password = empty($_SERVER['PHP_AUTH_PW']) ? NULL : $_SERVER['PHP_AUTH_PW'];
+
+    // Try to fill PHP_AUTH_USER & PHP_AUTH_PW with REDIRECT_HTTP_AUTHORIZATION
+    // for compatibility with Apache PHP CGI/FastCGI.
+    // This requires the following line in your ".htaccess"-File:
+    // RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
+    if (!empty($_SERVER['REDIRECT_HTTP_AUTHORIZATION']) && !isset($username) && !isset($password)) {
+      $authentication = base64_decode(substr($_SERVER['REDIRECT_HTTP_AUTHORIZATION'], 6));
+      list($username, $password) = explode(':', $authentication);
+      $_SERVER['PHP_AUTH_USER'] = $username;
+      $_SERVER['PHP_AUTH_PW'] = $password;
+    }
+
+    return array($username, $password);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getUser() {
+    list($user,) = static::getCredentials();
```

Comment:
$user => $account (as best practice, to avoid touching the ``global $user`` somewhere down the line).